File: python/mario.py
from cs50 import get_int


def main():

    # Prompt user for the pyramid height
    height = get_int("Height: ")

    # Set height limit
    while not height in range(0, 24):
        height = get_int("Height: ")

    # For each row
    for row in range(height):

        # Print spaces
        for space in range(height - row - 1):

            print(" ", end="")

        # Print left half
        for lefthash in range(height - row, height + 1):

            print("#", end="")

        # Print gap
        print("  ", end="")

        # Print right half
        for righthash in range(height - row, height + 1):

            print("#", end="")

        # Print a new line
        print()


if __name__ == "__main__":
    main()

File: programs/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Determine variables
    int height, line, space, hash;

    // Set height limit to 23
    do {
        printf("Height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);

    // Print lines based on height
    for (line = 1; line <= height; line++)
    {
        // Print spaces
        for (space = 0; space < (height - line); space++)
        {
            printf(" ");
        }

        // Print hashes on left
        for (hash = 0; hash < line; hash++)
        {
            printf("#");
        }

        // Print gap
        printf("  ");

        // Print hashes on right
        for (hash = 0; hash < line; hash++)
        {
            printf("#");
        }

        printf("\n");
    }
}